### PR TITLE
enhancement: Adding UI for variables of any json type

### PR DESCRIPTION
**Pull request by WillRaphaelson**
This PR:
- Updates create and edit modals to use the json input, which formats highlights and validates.
- Links the name to the opening of the edit modal
- Retains the value column for any variable with length <= 64 characters, and provides a link that opens the modal if its longer.
- Correct a bug that was causing the edit modal to display backticks.

For the team - should i be reaching back into the types more than i am?

image

image

All thoughts welcome, thanks.

### CODE REVIEW

**Review by collincchoy — COMMENTED** [review]
This looks good to me! Mostly non-blocking comments - only [1 comment I think might need to be addressed](https://github.com/PrefectHQ/prefect-ui-library/pull/2414#discussion_r1603648654) around handling a variable edit/update from a clicked VariableLink. I _think_(?) I'm right there that atm you'd update and not see your updated data. 

> should i be reaching back into the types more than i am?

Hm 🤔 not sure what you mean.

File: src/components/VariablesTable.vue [hunk 1]
```diff
@@ -32,10 +31,19 @@
     >
       <template #name="{ row }">
         <div class="variables-table__name" :title="row.name">
-          {{ row.name }}
+          <VariableLink :variable="row" />
```
[the remark this answers is not in the document]
Comment (collincchoy) [reply src/components/VariablesTable.vue:34]: I think you probably want something like `@update="refreshSubscriptions"` . ... or I guess the table has an event for updating also so maybe `@update="handleUpdate($event)"` is the more "correct" thing here.

File: src/components/VariablesTable.vue [hunk 2]
```diff
@@ -32,10 +31,19 @@
...
+        </div>
+      </template>
+
+
+      <template #value="{ row }">
+        <div class="variables-table__value" :title="row.name">
+          <p-code-highlight v-if="row.value.length < 64" :text="row.value" lang="json" inline />
```
[the remark this answers is not in the document]
Comment (WillRaphaelson) [reply src/components/VariablesTable.vue:41]: yeah it'd just wrap

**Comment by WillRaphaelson** [comment]
> This looks good to me! Mostly non-blocking comments - only [1 comment I think might need to be addressed](https://github.com/PrefectHQ/prefect-ui-library/pull/2414#discussion_r1603648654) around handling a variable edit/update from a clicked VariableLink. I _think_(?) I'm right there that atm you'd update and not see your updated data.
> 
> > should i be reaching back into the types more than i am?
> 
> Hm 🤔 not sure what you mean.

I mean like should i be changing the types for [variables](https://github.com/PrefectHQ/prefect-ui-library/blob/2a300917af88a2e45d9ce3669113e3fdf966b29d/src/models/Variable.ts#L6) from stringg to like string|int etc?

**Review by collincchoy — APPROVED** [review]
cha-ching

**Pull request merged**